### PR TITLE
PR: IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10 - Initial release - Fix 2833 

### PR DESCRIPTION
**Pull request by William-Francillette**
#### This Pull Request (PR) fixes the following issues

  * IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10
    * Initial release
      FIXES [#2833](https://github.com/microsoft/Microsoft365DSC/issues/2833)

### CODE REVIEW

**Comment by William-Francillette** [comment]
Added missing stub to fix unit test

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.psm1 [hunk 1]
```diff
@@ -0,0 +1,1707 @@
...
+            $enumPolicyType = $null
+            if ($null -ne $definition.PolicyType)
+            {
+                $enumPolicyType = $definition.PolicyType.ToString()
+            }
+            $complexDefinition = @{
+                CategoryPath = $definition.CategoryPath
+                ClassType    = $enumClassType
+                DisplayName  = $definition.DisplayName
+                #ExplainText           = $definition.ExplainText
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.psm1:150]: Why are these lines commented out? Can't they be removed altogether? The schema does have these parameters.

**Review by ykuijs — CHANGES_REQUESTED** [review]
One small question on some code that is commented out

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.psm1 (hunk 1 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10/MSFT_IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.psm1]: @ykuijs just removed those commented parameters
Most of those are added as read only just to give some context on the export

**Comment by ykuijs** [comment]
QQ: If these properties are Read-Only, is there a reason that they are still mentioned in the schema.mof file?

**Comment by William-Francillette** [comment]
> QQ: If these properties are Read-Only, is there a reason that they are still mentioned in the schema.mof file?

I didn't want them to generate an error in the export or the compilation 
They are returned in the export but are not tested and not added in the update. I just wanted the export to be more readable otherwise the setting would have been an id and not really readable. The definition parameter is not required for the resource but it gives all the information about the setting.

**Comment by William-Francillette** [comment]
This resource was quite a tricky one as hundreds of settings are available and they are split in definition and presentation Definition. 
I also used the rest query to update presentation value as it is not available in the sdk and will be released in 2.0 
This resource has been requested a few times so wanted to release it asap

**Review by ykuijs — APPROVED** [review]
LGTM

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+                            deviceAndAppManagementAssignmentFilterType = 'none'
+                            AdditionalProperties                       = @{'@odata.type' = '#microsoft.graph.allDevicesAssignmentTarget' }
+                        }
+                    })
+            }
+            Mock -CommandName Update-DeviceConfigurationPolicyAssignment -MockWith {
+            }
+            Mock -CommandName Update-DeviceConfigurationGroupPolicyDefinitionValue -MockWith {
+            }
+
```
Comment (ykuijs) [inline Tests/Unit/Microsoft365DSC/Microsoft365DSC.IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.Tests.ps1:58]: Please also add a Mock for Write-Host to suppress the output during the unit tests, like is done in other resources.

**Review by ykuijs — COMMENTED** [review]
Few updates required to fix the unit tests

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+                    Ensure                           = 'Present'
+                    Credential                       = $Credential
+                    Assignments                      = [CimInstance[]]@(
+                        (New-CimInstance -ClassName MSFT_DeviceManagementConfigurationPolicyAssignments -Property @{
+                            deviceAndAppManagementAssignmentFilterType = 'none'
+                            dataType                                   = '#microsoft.graph.allDevicesAssignmentTarget'
+                        } -ClientOnly)
+                    )
+                    DefinitionValues                 = [CimInstance[]]@(
+                        (N-ClientOnly)
```
[the remark this answers is not in the document]
Comment (William-Francillette) [reply Tests/Unit/Microsoft365DSC/Microsoft365DSC.IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.Tests.ps1]: Not sure where this came from but resolved now

File: Tests/Unit/Microsoft365DSC/Microsoft365DSC.IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.Tests.ps1 (hunk 2 above)
Comment (William-Francillette) [reply Tests/Unit/Microsoft365DSC/Microsoft365DSC.IntuneDeviceConfigurationAdministrativeTemplatePolicyWindows10.Tests.ps1:61]: Sure, done

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**